std-bits/base/src/main/java/org/enso/base/parser/FormatDetectingNumberParser.java

```diff
@@ -0,0 +1,301 @@
+package org.enso.base.parser;
+
+/**
+ * Parse a String into a Number. It supports the following patterns: - SIGN + NUMBER; - BRACKETS +
+ * SPACE + NUMBER + SPACE + BRACKET_CLOSE;
```

Comment:
It would be preferred to use the Javadoc syntax which borrows from HTML:
```suggestion
 * Parse a String into a Number. It supports the following patterns: 
 * <ul>
 *   <li>SIGN + NUMBER</li>
 *   <li>BRACKETS + SPACE + NUMBER + SPACE + BRACKET_CLOSE</li>
 * </ul>
```